test/clj/rems/test_form_validation.clj

```diff
@@ -218,6 +234,18 @@
                                            :field/visible  false
                                            :field/value    ""}]))))
 
+  (testing "invisible fields should not be filled"
```

Comment:
both tests suits have the same name? ```"invisible fields should not be filled"```

Reply:
Heh, good catch! The first one is supposed to test `validate-fields-for-submit` and the second one `validate-fields-for-draft` but it looks like they both test `-draft`. I'll fix it.

Reply:
👍🏻

Reply:
Otherwise, it looks alright, I think that form logic is complex in any case so suggestions to simplify hardly would make any sense coming from me